### PR TITLE
feat(kprompt): introduce kui tokens [KHCP-7719]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-7719

Changes:
- introduces `kui-` design tokens in `KPrompt` component
- removes any usage of Kongponents utility classes in `KPrompt` component template

## PR Checklist

* [ ] Does not introduce dependencies
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KPrompt/KPrompt.vue [hunk 1]
```diff
@@ -7,29 +7,28 @@
...
-        <div class="k-prompt-header-content d-flex align-items-center w-100">
+      <div class="k-prompt-header">
+        <div class="k-prompt-header-content">
           <slot name="header-content">
             <KIcon
               v-if="type === 'warning'"
-              class="mr-2"
-              color="var(--white)"
+              class="warning-icon"
+              :color="`var(--white, ${KUI_COLOR_TEXT_INVERSE})`"
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:16]: Do we need the CSS variable token here? 
```suggestion
              :color="`var(--white, var(--kui-color-text-inverse, ${KUI_COLOR_TEXT_INVERSE}))`"
```

File: src/components/KPrompt/KPrompt.vue [hunk 2]
```diff
@@ -7,29 +7,28 @@
...
               icon="warning"
-              secondary-color="var(--yellow-400)"
-              size="20"
+              secondary-color="currentColor"
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:18]: issue: I'm not sure this makes sense; this means the color of the parent would have to be yellow

File: src/components/KPrompt/KPrompt.vue [hunk 3]
```diff
@@ -7,29 +7,28 @@
...
           </slot>
           <div class="close-button">
             <KButton
               aria-label="Close"
-              class="non-visual-button"
               @click="close"
             >
               <KIcon
-                color="var(--grey-600)"
+                :color="`var(--grey-600, ${KUI_COLOR_TEXT_NEUTRAL})`"
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:29]: same comment here about providing the CSS variable token

File: src/components/KPrompt/KPrompt.vue [hunk 4]
```diff
@@ -81,9 +79,9 @@
             <template #icon>
               <KIcon
                 v-if="actionPending"
-                color="var(--grey-400)"
+                :color="`var(--grey-400, ${KUI_COLOR_TEXT_NEUTRAL_WEAK})`"
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:82]: same comment here about CSS variable

File: src/components/KPrompt/KPrompt.vue [hunk 5]
```diff
@@ -227,62 +226,92 @@ onBeforeUnmount(() => {
...
+            color: var(--grey-600, var(--kui-color-text-neutral-strong, $kui-color-text-neutral-strong));
+            font-size: var(--type-md, var(--kui-font-size-40, $kui-font-size-40));
+            line-height: var(--kui-line-height-40, $kui-line-height-40);
+            max-height: var(--KPromptMaxHeight, 300px);
+            overflow-x: hidden;
+            overflow-y: auto;
+            padding-bottom: var(--spacing-md, var(--kui-space-60, $kui-space-60));
+            text-align: start;
+            white-space: normal; // in case inside KTable
+            width: 100% !important;
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:297]: question: Why did the width change from `99%` to `100%`?

File: src/components/KPrompt/KPrompt.vue (hunk 2 above)
Comment (portikM) [reply src/components/KPrompt/KPrompt.vue:18]: No, I set color for the `.warning-icon` class I've given this element

File: src/components/KPrompt/KPrompt.vue (hunk 5 above)
Comment (portikM) [reply src/components/KPrompt/KPrompt.vue:297]: We had `.w-100` utility class on this element which was clearly superseding this width rule.

File: src/components/KPrompt/KPrompt.vue (hunk 1 above)
Comment (portikM) [reply src/components/KPrompt/KPrompt.vue]: Fixed

File: src/components/KPrompt/KPrompt.vue (hunk 3 above)
Comment (portikM) [reply src/components/KPrompt/KPrompt.vue]: Fixed

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**